From 902dc2cb203c495ce45f1003c5188c4b6dca4bf8 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 4 Mar 2011 11:34:44 +0100
Subject: [PATCH] base/dracut-lib.sh: fix "die()"

die() now enters emergency mode and displays message there after
plymouth was stopped.
---
 modules.d/99base/dracut-lib.sh | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/modules.d/99base/dracut-lib.sh b/modules.d/99base/dracut-lib.sh
index 9666232..5ae86a8 100644
--- a/modules.d/99base/dracut-lib.sh
+++ b/modules.d/99base/dracut-lib.sh
@@ -87,10 +87,12 @@ die() {
     } > /dev/kmsg
 
     { 
-        echo "dracut: FATAL: $@";
-        echo "dracut: Refusing to continue";
-    } >&2
-    
+        echo "warn dracut: FATAL: \"$@\"";
+        echo "warn dracut: Refusing to continue";
+	echo "exit 1"
+    } >> /emergency/01-die.sh
+
+    > /.die
     exit 1
 }
 
